cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/planner/ConnectorDag.java

```diff
@@ -227,6 +228,64 @@ private void isolate(String node, Set<String> addedAlready) {
   }
 
   /**
+   * This method first split the Dag based on the input set of conditions.
+   * The individual splitted dag then further split by using {@link #split} method based on the reduce nodes.
+   * @param conditionNodes Set of condition nodes based on which dag is to be pre-split.
+   * @return the subdags.
+   */
+  public List<Dag> splitByConditions(Set<String> conditionNodes) {
+    List<Dag> dags = new ArrayList<>();
+    Set<String> parentStopperNodes = Sets.union(sources, conditionNodes);
+    Set<String> childStopperNodes = Sets.union(sinks, conditionNodes);
+    for (String condition : conditionNodes) {
```

Comment:
it would be useful to add a comment before this loop with an example dag (for example, one from a test case), then comment on what the subdags should be after this loop.